PWGHF/Core/HfMlResponseB0ToDPi.h

```diff
@@ -130,11 +140,76 @@ class HfMlResponseB0ToDPi : public HfMlResponse<TypeOutputScore>
         CHECK_AND_FILL_VEC_B0(cpaXY);
         CHECK_AND_FILL_VEC_B0(maxNormalisedDeltaIP);
         // TPC PID variable
-        CHECK_AND_FILL_VEC_B0_FULL(prong1, tpcNSigmaPi1, tpcNSigmaPi);
+        CHECK_AND_FILL_VEC_B0_FULL(prongBachPi, tpcNSigmaPi1, tpcNSigmaPi);
         // TOF PID variable
-        CHECK_AND_FILL_VEC_B0_FULL(prong1, tofNSigmaPi1, tofNSigmaPi);
+        CHECK_AND_FILL_VEC_B0_FULL(prongBachPi, tofNSigmaPi1, tofNSigmaPi);
         // Combined PID variables
-        CHECK_AND_FILL_VEC_B0_FUNC(prong1, tpcTofNSigmaPi1, o2::pid_tpc_tof_utils::getTpcTofNSigmaPi1);
+        CHECK_AND_FILL_VEC_B0_FUNC(prongBachPi, tpcTofNSigmaPi1, o2::pid_tpc_tof_utils::getTpcTofNSigmaPi1);
+      }
+      if constexpr (withDmesMl) {
+        if constexpr (reduced) {
+          switch (idx) {
+            CHECK_AND_FILL_VEC_B0(prong0MlScoreBkg);
+            CHECK_AND_FILL_VEC_B0(prong0MlScorePrompt);
+            CHECK_AND_FILL_VEC_B0(prong0MlScoreNonprompt);
+          }
+        } else {
+          if (mlScoresD) {
+            switch (idx) {
+              CHECK_AND_FILL_VEC_B0_INDEX(prong0MlScoreBkg, *mlScoresD, 0);
+              CHECK_AND_FILL_VEC_B0_INDEX(prong0MlScorePrompt, *mlScoresD, 1);
+              CHECK_AND_FILL_VEC_B0_INDEX(prong0MlScoreNonprompt, *mlScoresD, 2);
+            }
+          } else {
+            LOG(fatal) << "ML scores of D not provided";
+          }
+        }
+      }
+    }
+
+    return inputFeatures;
+  }
+
+  /// Method to get the input features vector needed for ML inference
+  /// \param candidate is the B0 candidate
+  /// \param prongBachPi is the candidate's bachelor pion prong
+  /// \param mlScoresD is the vector of ML scores for the D meson (if available)
+  /// \note this method is used for B0 → D*∓ π± candidates with D meson ML scores
+  /// \return inputFeatures vector
+  template <bool withDmesMl, IsB0ToDstarPiChannel T1, typename T2>
+  std::vector<float> getInputFeatures(T1 const& candidate,
+                                      T2 const& prongBachPi,
+                                      const std::vector<float>* mlScoresD = nullptr)
+  {
+    std::vector<float> inputFeatures;
+    auto prongSoftPi = candidate.template prongSoftPi_as<o2::soa::Join<o2::aod::HfRedSoftPiBases, o2::aod::HfRedSoftPiCov, o2::aod::HfRedSoftPiPid>>();
```

Comment:
Sorry, I don't get what you are trying to do here.
What's the point of overloading the existing `getInputFeatures` function with another one which behaves differently?
Why don't you take `prongSoftPi` as the function argument?